hack/tools/tilt-prepare/main.go

```diff
@@ -72,8 +75,23 @@ var (
 	kustomizeBuildsFlag  = pflag.StringSlice("kustomize-builds", []string{}, "list of kustomize build to be run; each value should be in the form name:path")
 	providersBuildsFlag  = pflag.StringSlice("providers", []string{}, "list of providers to be installed; each value should be in the form name:path[:debug]")
 	allowK8SContextsFlag = pflag.StringSlice("allow-k8s-contexts", []string{}, "Specifies that Tilt is allowed to run against the specified k8s context name; Kind is automatically allowed")
+	tiltSettingsFileFlag = pflag.String("tilt-settings-file", "", "Path to a tilt-settings.(json|yaml) file")
 )
 
+type debugConfig struct {
+	Continue     *bool `json:"continue"`
+	Port         int   `json:"port"`
+	ProfilerPort int   `json:"profiler_port"`
+	MetricsPort  int   `json:"metrics_port"`
+}
+
+type extraArgs []string
+
+type tiltSettings struct {
+	Debug     map[string]*debugConfig `json:"debug,omitempty"`
```

Comment:
Make provider keys real types vs map[string]? Potential downside of former: We have to hard-code and keep up with all external providers?!

Reply:
We can't keep up with external providers as some of them are not public and should still be supported :)

So I think a string as key is totally fine.